test/Microsoft.Extensions.DataIngestion.Tests/MakdownChunker/MarkdownChunker.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.DataIngestion.Tests
+{
+    public sealed class MarkdownChunker : DocumentChunker
+    {
+        private readonly int _headerLevelToSplitOn;
+        private readonly bool _stripHeaders;
+
+
+        public MarkdownChunker(MarkdownHeaderLevel HeaderLevelToSplitOn = MarkdownHeaderLevel.Header3, bool StripHeaders = true)
+        {
+            _headerLevelToSplitOn = (int)HeaderLevelToSplitOn;
+            _stripHeaders = StripHeaders;
+        }
+
+        public override ValueTask<List<Chunk>> ProcessAsync(Document document, CancellationToken cancellationToken = default)
+        {
+            if (document is null) throw new ArgumentNullException(nameof(document));
+
+            string markdown = document.Markdown.ReplaceLineEndings();
```

Comment:
I've created https://github.com/adamsitnik/dataingestion/issues/68 to ensure we solve the new lines problem in the right way.